pkgs/development/tools/godot/default.nix

```diff
@@ -19,30 +19,48 @@
 , libGLU
 , zlib
 , yasm
-, withUdev ? true
 }:
 
-let
-  options = {
-    touch = libXi != null;
-    pulseaudio = false;
-    udev = withUdev;
-  };
-in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (self:
+{
   pname = "godot";
   version = "3.4.4";
 
+  name = "${self.pname}-${self.version}";
+
   src = fetchFromGitHub {
     owner = "godotengine";
     repo = "godot";
-    rev = "${version}-stable";
-    sha256 = "sha256-3AESLzqozi7Fc80u8Ml3ergZMkIhHy4tNlRe/3FsE6k=";
+    rev = self.passthru.githubRev;
+    sha256 = self.passthru.githubSha256;
   };
 
-  nativeBuildInputs = [ pkg-config ];
-  buildInputs = [
+  passthru = {
+    githubRev = "${self.version}-stable";
+    githubSha256 = "sha256-3AESLzqozi7Fc80u8Ml3ergZMkIhHy4tNlRe/3FsE6k=";
```

Comment:
What do you mean by putting them "in overrides"? Overrides of what?

Reply:
It should be set in inputs but I don't think we even need that if we use a full fetchFromGitHub in the other packages which is also was more flexible.

Reply:
The purpose of having the github rev and hash in passthru is so that a package defined through `godot.overrideAttrs` can override them.  Without this, they would have to override `src` with a full fetchFromGitHub call.  This allows the override to say "use the same GitHub source as the base package, but a different revision", without repeating the owner and repo.  Also it doesn't *prevent* the overriding package from using a full fetchFromGitHub call:  it can still override src if it wants to change to a completely different way of obtaining the source.

Reply:
I still don't like this. Changing variables in passthru should not have any effect on the package. This should all be moved to inputs and only inherit things to be consumed by downstream things.

Also repeating things in source is not a big deal. I'd prefer it was more than the current more complicated approach.